src/components/Posts/PostMenu.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  IconButton,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  MenuItem,
+} from "@mui/material";
+import React, { useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { MoreVert, Edit, Delete, Link } from "@mui/icons-material";
+import axios from "axios";
+import { deletePost } from "../../state/slices/postsSlice";
+import { useNavigate } from "react-router-dom";
+
+type PostMenuProps = {
+  authorId: number;
+  postId: number;
+  isExpanded?: boolean;
+};
+
+const styles = {
+  listItemIcon: {
+    "&.MuiListItemIcon-root": { color: "black.main" },
+  },
+  listItemText: { fontWeight: "bold" },
+  menu: {
+    borderRadius: 4,
+  },
+};
+
+const PostMenu = ({ authorId, postId, isExpanded = false }: PostMenuProps) => {
+  const userId = useAppSelector((state) => state.user.userId);
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const handleDelete = async () => {
+    try {
+      const result = await axios.delete(
+        `http://localhost:3001/api/posts/deletePost`,
+        {
+          data: {
+            postId: postId,
+            userId: userId,
+          },
+        }
+      );
+
+      isExpanded ? handleExpandedPostDelete() : handlePostDelete();
```

Comment:
My personal preference is to inline code whenever we can because it's easier to read than to jump around function definitions. Since this code isn't overly complex, can we put the code paths inside the function instead? I also don't think we need to return the data here since it got deleted/the state got updated in the store. Something like
```suggestion
      if(isExpanded) {
        dispatch(deletePost(postId));
      }
      else {
        navigate(-1)
      }
      setMenuOpen(false);
```

Reply:
I modified it to inline code for the if statement, but I kept the menu closing inside of the finally block with a separate function definition since I handle closing the modal in other elements as well

Reply:
Nice I like this better